# papers-for-AMP-using-Deeplearning

## Paper and Code

|   Date   | Title |  Conf./Journal  | Code  |
| :------: | :-----------------: | :-----------------------: | :----------------------: |
|  2021 | A novel antibacterial peptide recognition algorithm based on BERT|     [Briefings in Bioinformatics](https://academic.oup.com/bib/article-abstract/22/6/bbab200/6284370?redirectedFrom=fulltext)     |      [code](https://github.com/BioSequenceAnalysis/Bert-Protein)     |
|  2022 | Accelerating the discovery of antifungal peptides using deep temporal convolutional networks|     [Briefings in Bioinformatics](https://academic.oup.com/bib/article-abstract/23/2/bbac008/6526725?redirectedFrom=fulltext)     |      [code](https://tcn-afppred.anvil.app/)     |
|  2020 | ACEP: improving antimicrobial peptides recognition through automatic feature fusion and amino acid embedding|     [BMC Genomics](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-06978-0)     |      [code](https://github.com/Fuhaoyi/ACEP)     |
|  2021 | AI4AMP: an Antimicrobial Peptide Predictor Using Physicochemical Property-Based Encoding Method and Deep Learning|     [mSystems](https://journals.asm.org/doi/10.1128/mSystems.00299-21)     |      [code](https://github.com/LinTzuTang/AI4AMP_predictor)     |
|  2022 | AMP-BERT: Prediction of antimicrobial peptide function based on a BERT model|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/GIST-CSBL/AMP-BERT)     |
|  2022 | AMPDeep: hemolytic activity prediction  of antimicrobial peptides using transfer learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/milad73s/AMPDeep)     |
|  2021 | AniAMPpred: artificial intelligence guided discovery of novel antimicrobial peptides in animal kingdom|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://aniamppred.anvil.app/)     |
|  2022 | Identification of antimicrobial peptides from the  human gut microbiome using deep learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/mayuefine/c_AMPs-prediction)     |
|  2019 | Antimicrobial peptide identification using multi-scale convolutional network|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/zhanglabNKU/APIN)     |
|  2022 | Antimicrobial Peptides Prediction method based on sequence multidimensional feature embedding|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |   None  |
|  2021 | Deep-ABPpred: identifying antibacterial peptides in protein sequences using bidirectional LSTM with word2vec|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://abppred.anvil.app/)     |
|  2021 | Deep Learning for Novel Antimicrobial Peptide Design|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://www.mdpi.com/2218-273X/11/3/471/s1)     |
|  2018 | Deep learning improves antimicrobial peptide recognition|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code]( www.ampscanner.com)     |
|  2020 | Deep-AmPEP30: Improve Short Antimicrobial Peptides Prediction with Deep Learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://cbbio.cis.um.edu.mo/AxPEP)     |
|  2020 | Design, Screening, and Testing of Non-Rational Peptide Libraries with Antimicrobial Activity: In Silico and Experimental Approaches|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |  None   |
|  2022 | Handcrafted versus non-handcrafted (self-supervised) features for the classification of antimicrobial peptides: complementary or redundant?|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None  |
|  2021 | iAMP-CA2L: a new CNN-BiLSTM-SVM classifier based on cellular automata image for identifying antimicrobial peptides and their functional types|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/liujin66/iAMP-CA2L)     |
|  2022 | Identifying novel antimicrobial peptides from venom gland of spider Pardosa astrigera by deep multi-task learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/bzlee-bio/AMPSpeciesSpecific)     |
|  2022 | Integrating transformer and imbalanced multi-label learning to identify antimicrobial peptides and their functional activities|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/BiOmicsLab/TransImbAMP)     |
|  2022 | LABAMPsGCN: A framework for identifying lactic acid bacteria antimicrobial peptides based on graph convolutional neural network|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None   |
|  2022 | AMPlify: attentive deep learning model  for discovery of novel antimicrobial peptides  efective against WHO priority pathogens|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/bcgsc/AMPlify)     |
|  2022 |Prediction Models for Identifying Ion Channel-Modulating Peptides via Knowledge Transfer Approaches|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/bzlee-bio/PrIMP)     |
|  2022 | Rational Discovery of Antimicrobial Peptides by Means of Artificial Intelligence|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/BCV-Uniandes/AMPs-Net)     |
|  2021 | StaBle-ABPpred: a stacked ensemble predictor based on biLSTM and attention mechanism for accelerated discovery of antibacterial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://stable-abppred.anvil.app/)     |
|  2018 | AmPEP: sequence-based prediction of antimicrobial peptides using distribution patterns of amino acid properties and random forest|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2018 | Computational antimicrobial peptide design and evaluation against multidrug-resistant clinical isolates of bacteria|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/Tushar-N/amp-lm)     |
|  2017 | Predicting antimicrobial peptides with improved accuracy by incorporating the compositional, physico-chemical and structural features into Chouâ€™s general PseAAC|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://cabgrid.res.in:8080/amppred/)     |
|  2013 | iAMP-2L: a two-level multi-label classifier for identifying antimicrobial peptides and their functional types|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://www.jci-bioinfo.cn/iAMP-2L)     |
|  2022 |Do deep learning models make a difference in the identification of antimicrobial peptides?|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2022 | LMPred: predicting antimicrobial peptides using pre-trained language models and deep learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/williamdee1/)     |
|  2021 | Construction and Prediction of Antimicrobial Peptide Predicition Model Based on BERT|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2022 | Benchmarks in antimicrobial peptide prediction are biased due to the selection of negative data|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://BioGenies.info/AMPBenchmark)     |
|  2022 |AMPpred-EL: An effective antimicrobial peptide prediction model based on ensemble learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None   |
|  2022 | ProPythia: a Python package for protein classification based on machine and deep learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/BioSystemsUM/propythia)     |
|  2022 | Sense the moment: A highly sensitive antimicrobial activity predictor based on hydrophobic moment|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/williamfp7/stm)     |
|  2023 | sAMPpred-GAT: prediction of antimicrobial peptide by graph attention network and predicted peptide structure|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/HongWuL/sAMPpred-GAT/)     |
|  2021 | Hmd-amp: Protein language-powered hierarchical multi-label deep forest for annotating antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None   |
|  2022 | Coupled encoding methods for antimicrobial peptide prediction: how sensitive is a highly accurate model?|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None  |
|  2020 | IAMPE: NMR-assisted computational prediction of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://cbb1.ut.ac.ir/)     |
|  2022 |Identifying multi-functional bioactive peptide functions using multi-label deep learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/xialab-ahu/MLBP)     |
|  2022 | sAMP-PFPDeep: Improving accuracy of short antimicrobial peptides prediction using three different sequence encodings and deep neural networks|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/WaqarHusain/sAMP-PFPDeep)     |
|  2021 | Machine learning-enabled predictive modeling to precisely identify the antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None   |
|  2019 | An advanced approach to identify antimicrobial peptides and their function types for penaeus through machine learning strategies|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None   |
|  2020 | Alphabet reduction and distributed vector representation based method for classification of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None  |
|  2020 | Characterization and identification of antimicrobial peptides with different functional activities|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://fdblab.csie.ncu.edu.tw/AMPfun/index.html)     |
|  2023 | AFP-MFL: accurate identification of antifungal peptides using multi-view feature learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://inner.wei-group.net/AFPMFL/)     |
|  2019 | Identifying antimicrobial peptides using word embedding with deep recurrent neural networks|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/nafizh/NeuBI)     |
|  2022 | Deep-AFPpred: identifying novel antifungal peptides using pretrained embeddings from seq2vec with 1DCNN-BiLSTM|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://afppred.anvil)     |
|  2020 | ampir: an R package for fast genome-wide prediction of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code]( https://github.com/legana/ampir)     |
|  2021 | Using an ensemble to identify and classify macroalgae antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None  |
|  2021 | Co-AMPpred for in silico-aided predictions of antimicrobial peptides by integrating composition-based features|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/onkarS23/CoAMPpred)     |
|  2021 | Ensemble-AMPPred: robust AMP prediction and recognition using the ensemble learning method with a new hybrid feature for differentiating AMPs|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None  |
|  2020 |Macrel: antimicrobial peptide screening in genomes and metagenomes|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/BigDataBiology/macrel)     |
|  2020 | Proteomic screening for prediction and design of antimicrobial peptides with AmpGram|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/michbur/AmpGram-analysis)     |
|  2022 |iAFPs-EnC-GA: identifying antifungal peptides using sequential and evolutionary descriptors based multi-information fusion and ensemble learning approach|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/farmanit335/iAFPs-EnC-GA)     |
|  2019 | Toward insights on determining factors for high activity in antimicrobial peptides via machine learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2022 | ABP-Finder: A Tool to Identify Antibacterial Peptides and the Gram-Staining Type of Targeted Bacteria|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://protdcal.zmb.uni-due.de/ABP-Finder/)     |
|  2018 | Novel 3D Structure Based Model for Activity Prediction and Design of Antimicrobial Peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2019 | Deep learning regression model for antimicrobial peptide design|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/zswitten/Antimicrobial-Peptides)     |
|  2019 | De novo design and in vitro testing of antimicrobial peptides against Gram-negative bacteria|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://dbaasp.org/prediction)     |
|  2023 | Discovering highly potent antimicrobial peptides with deep generative model HydrAMP|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/szczurek-lab/hydramp)     |
|  2022 | Feedback-AVPGAN: Feedback-guided generative adversarial network for generating antiviral peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/KanoHase/AVP-Generator)     |
|  2020 | Variational Autoencoder for Generation of Antimicrobial Peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None   |
|  2021 | Combining genetic algorithm with machine learning strategies for designing potent antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None    |
|  2020 | Generating ampicillin-level antimicrobial peptides with activity-aware generative adversarial networks|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/tsudalab/PepGAN)     |
|  2023 | Latent Spaces for Antimicrobial Peptide Design|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/Mansbach-Lab/latent-spaces-amps)     |
|  2021 | AMPGAN v2: machine learning-guided design of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://gitlab.com/vail-uvm/amp-gan)     |
|  2022 | Accelerating Antimicrobial Peptide Discovery with Latent Sequence-Structure Model|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/dqwang122/LSSAMP)     |
|  2021 | Discovering Novel Antimicrobial Peptides in Generative Adversarial Network|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/lsbnb/amp_gan)     |
|  2021 | Accelerated antimicrobial discovery via deep generative models and molecular dynamics simulations|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/IBM/controlled-peptide-generation)     |
|  2023 | Application of a deep generative model produces novel and diverse functional peptides against microbial resistance|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/AspirinCode/AMPTrans-lstm)     |
|  2022 | Deep Learning-Based Bioactive Therapeutic Peptide Generation and Screening|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/haiping1010/New_peptide_iteration)     |
|  2022 | Deep attention based variational autoencoder for antimicrobial peptide discovery|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/ghorbanimahdi73/AMPGen)     |
|  2021 | Machine learning designs non-hemolytic antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2022 | Cell-free biosynthesis combined with deep learning accelerates de novo-development of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/amirpandi/Deep_cAMP)     |
|  2021 | PepVAE: variational autoencoder framework for antimicrobial peptide generation and activity prediction|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      None   |
|  2023 | Reprogramming Pretrained Language Models for Protein Sequence Representation Learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://github.com/riavinod/r2dl)     |
|  2020 | Deep learning enables the design of functional de novo antimicrobial proteins|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None   |
|  2021 | Alignment-free antimicrobial peptide predictors: improving performance by a thorough analysis of the largest available data set|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](https://biocom-ampdiscover.cicese.mx/)     |
|  2019 | Prediction of antifungal peptides by deep learning with character embedding|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |      [code](http://vivace.bi.a.u-tokyo.ac.jp:8008/fang/AFPDeep.php)     |
|  2021 |Comprehensive assessment of machine learning-based methods for predicting antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |   None    |
|  2019 |Encodings and models for antimicrobial peptide classification for multi-resistant pathogens|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None     |
|  2022 | Recent Progress in the Discovery and Design of Antimicrobial Peptides Using Traditional Machine Learning and Deep Learning|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None     |
|  2021 | Sequence-based peptide identification, generation, and property prediction with deep learning: A review|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |   None     |
|  2022 | A review on antimicrobial peptides databases and the computational tools|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None     |
|  2021 | Antimicrobial peptides: An update on classifications and databases|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None     |
|  2022 | Synthetic molecular evolution of antimicrobial peptides|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None    |
|  2021 | Synthetic biology and computer-based frameworks for antimicrobial peptide discovery|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |   None    |
|  2021 | Computational methods and tools in antimicrobial peptide research|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None    |
|  2022 | Emerging Computational Approaches for Antimicrobial Peptide Discovery|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None    |
|  2021 | A review of artificial intelligence applications for antimicrobial resistance|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None      |
|  2022 | De novo peptide and protein design using generative adversarial networks: an update|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None       |
|  2022 | Deep generative models for peptide design|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |    None     |
|  2023 | Identification of potent antimicrobial peptides via a machine-learning pipeline that mines the entire space of peptide sequences|     [Protein Science](https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.4529)     |     None   |
